### ISSUE TITLE
BT output: Only noise when playing local files or internet radio

### ISSUE THREAD

**Issue by hvonbargen** [mopidy/mopidy#1411]
Strange: I have set up Mopidy on Raspberry Pi Model B.
I have connected an USB BT dongle to send sound output to a Philips BT connected to my hifi-receiver.
I can listen to Spotify using mopidy fine.
But, trying to stream internet radio (any of the samples on the musicbox_webclient streams page) or trying to play a local mp3 file, I hear only noise.
Whereas mplayer -ao alsa:device=philipsbt http://...../1live-diggi is working fine.

This is my /etc/asound.conf (note: I edited the file without actually understanding it):

pcm.mmap0 {
    type mmap_emul;
    slave {
      pcm "hw:0,0";
    }
}

pcm.philipsbt {
   type plug
   slave {
       pcm {
           type bluetooth
           device 0C:A6:94:3D:6D:CA
           profile "auto"
       }   
   }   
   hint {
       show on
       description "BT Headset PhilipsBT"
   }   
}

ctl.bhilipsbt {
  type bluetooth
} 

pcm.!default {
  type plug;
  slave {
    pcm philipsbt;
  }
}

ctl.!default {
  type plug;
  slave {
    ctl philipsbt;
  }
}

**Comment by jodal** [comment]
A while ago we changed the contribution guidelines to make it clear that GitHub Issues is for bugs and well-defined feature requests only, ref. https://docs.mopidy.com/en/latest/contributing/#issue-guidelines

I don't want to kill your discussion. I just want to make the issue tracker a useful development tool by removing as many non-actionable issues as possible.

Can you please continue the discussion at https://discuss.mopidy.com? Feel free to post another comment here with the link to your continued discussion.

**Comment by hvonbargen** [comment]
After 1 week I there was  no response on https://discuss.mopidy.com/t/bt-output-only-white-noise-when-playing-local-files-or-internet-radio/1076...

**Comment by hvonbargen** [comment]
I found the solution in a comment by liamw9534 on issue 705:

I had to configure the audio output differently:

_Not_ working configuration (except for Spotify!):

```
[audio]
output = alsasink
```
Working configuration:

```
[audio]
output = sbcenc ! a2dpsink device=philipsbt
```

It was a hard time finding the solution, though.

With no help from the community, I had to dive into the source code to find out _how_ the audio output  
is implemented, which led to gst aka gstreamer(?). I reckoned that the spotify plugin used different options for streaming than the local file plugin. Fortunately, before going into a debugging session, I searched the internet for how to test gst from the command line. Finally, I could reproduce the noise output with this command line:

gst-launch-0.10 filesrc location=~/music/The\ Cure/The\ Head\ on\ the\ Door/07\ -\ The\ Cure\ -\ Close\ To\ Me.mp3  '!' decodebin '!' audioconvert '!' audioresample '!' alsasink

Then further internet research led to issue 705 with the golden hint from Liam.

And so I tried this, resulting in wonderful music:

gst-launch-0.10 filesrc location=~/music/The\ Cure/The\ Head\ on\ the\ Door/07\ -\ The\ Cure\ -\ Close\ To\ Me.mp3  '!' decodebin '!' audioconvert '!' audioresample '!' sbcenc '!' a2dpsink device=aa

Interestingly, the value of device=xxx doesn't seem to matter, but the device option _is_ necessary (well, not tested with mopidy, only tested using gst-launch-0.1.0, withoud device=... it would say ERROR: Pipeline doesn't want to pause.).
